dubbo-common/src/main/java/org/apache/dubbo/common/utils/UrlUtils.java

```diff
@@ -445,6 +451,22 @@ public static boolean isServiceKeyMatch(URL pattern, URL value) {
         return urls.stream().filter(predicate).collect(Collectors.toList());
     }
 
+    public static boolean isConfigure(URL url) {
+        return CONFIGURATORS_CATEGORY.equals(url.getParameter(Constants.CATEGORY_KEY, Constants.DEFAULT_CATEGORY))
+                || OVERRIDE_PROTOCOL.equals(url.getProtocol());
+    }
+
+    public static boolean isRoute(URL url) {
+        return ROUTE_PROTOCOL.equals(url.getProtocol())
+                || ROUTERS_CATEGORY.equals(url.getParameter(Constants.CATEGORY_KEY, Constants.DEFAULT_CATEGORY));
+    }
+
+    public static boolean isProvider(URL url) {
+        return PROVIDERS_CATEGORY.equals(url.getParameter(Constants.CATEGORY_KEY, PROVIDERS_CATEGORY))
+                && !OVERRIDE_PROTOCOL.equals(url.getProtocol())
+                && !ROUTE_PROTOCOL.equals(url.getProtocol());

```

Comment:
may be we can apply the same approach here.

Reply:
agree.